server/src/main/java/com/paypal/selion/pojos/BrowserInformationCache.java

```diff
@@ -0,0 +1,242 @@
+        TestSlotInformation testSlotInformation = (nodeMap.get(url) == null) ? new TestSlotInformation() : nodeMap
+                .get(url);
+        testSlotInformation.addBrowserInfo(browserInformation);
+        if (nodeMap.get(url) == null) {
+            logger.log(Level.FINE, "Creating new entry -> " + url + " : [" + browserName + ":" + maxInstances + "]");
+            nodeMap.put(url, testSlotInformation);
+        } else {
+            logger.log(Level.FINE, "Added entry -> " + url + " : " + " : [" + browserName + ":" + maxInstances + "]");
+        }
+    }
+
+    /**
+     * Returns the total instances of a particular browser, available through all nodes. This methods takes an instance
+     * of {@link Registry} to clean the cache before returning the results.
+     * 
+     * @param browserName
+     *            Browser name as {@link String}
+     * @param registry
+     *            {@link Registry} instance.
+     * @return Total instances of a particular browser across nodes.
+     */
+    public synchronized int getTotalBrowserCapacity(String browserName, Registry registry) {
+        logger.entering(new Object[] { browserName, registry });
+        cleanCacheUsingRegistry(registry);
+        int totalBrowserCounts = 0;
+        for (Map.Entry<URL, TestSlotInformation> entry : nodeMap.entrySet()) {
+            BrowserInformation browserInfo = entry.getValue().getBrowserInfo(browserName);
+            totalBrowserCounts += (browserInfo == null) ? 0 : browserInfo.getMaxInstances();
+        }
+        logger.exiting(totalBrowserCounts);
+        return totalBrowserCounts;
+    }
+
+    private void cleanCacheUsingRegistry(Registry registry) {
+        List<URL> relevantURLs = getRegistryURLs(registry);
+        removeIrrelevantURLs(relevantURLs);
+    }
+
+    private List<URL> getRegistryURLs(Registry registry) {
+        Iterator<RemoteProxy> remoteProxyIterator = registry.getAllProxies().iterator();
+        List<URL> urlList = new ArrayList<URL>();
+        while (remoteProxyIterator.hasNext()) {
+            RemoteProxy remoteProxy = remoteProxyIterator.next();
+            urlList.add(remoteProxy.getRemoteHost());
+        }
+        return urlList;
+    }
+
+    private void removeIrrelevantURLs(List<URL> hotURLList) {
+        Iterator<URL> urlIterator = nodeMap.keySet().iterator();
+        while (urlIterator.hasNext()) {
+            URL cacheURL = urlIterator.next();
+            if (!hotURLList.contains(cacheURL)) {
+                nodeMap.remove(cacheURL);
+            }
+        }
+    }
+
+    /**
+     * <code>TestSlotInformation</code> holds the {@link BrowserInformation} corresponding to a individual unique test
+     * slot.
+     */
+    private static class TestSlotInformation {
+
+        private static final SeLionGridLogger logger = SeLionGridLogger.getLogger(TestSlotInformation.class);
+
+        // Browser information set.
+        private Set<BrowserInformation> browserInformationSet = null;
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Avoid using redundant field initializer for 'browserInformationSet'](https://www.codacy.com/app/SeLion/paypal-SeLion/file/2103178711/issues/source?bid=2928425&fileBranchId=3143459#l141)